low-level/sensors/impl/centos_7/systemd_watchdog.py

```diff
@@ -1415,9 +1414,34 @@ def _get_drive_fault_info(self, path):
             cmd = f"sudo smartctl -d scsi -A {self._drive_by_device_name[path]}"
         else:
             cmd = f"sudo smartctl -A {self._drive_by_device_name[path]}"
-        response, _ = self._run_command(cmd)
+        response, _, _ = self._run_command(cmd)
         return response
 
+    def _is_local_drive(self, object_path):
+        """
+        Detect Node server local drives using Hdparm tool.
+        Hdparm tool give information only for node drive.
+        For external JBOD/virtual drives it will give output as:
+        "SG_IO: bad/missing sense data "
+        Hdparm does not have support for NVME drives, for this drives it gives o/p as:
+        "failed: Inappropriate ioctl for device"
+        """
+        ENCL_DISK_ERR = "SG_IO: bad/missing sense data"

```

Comment:
>DISK_ERR_MISSING_SENSE_DATA = "SG_IO: bad/missing sense data"
>DISK_ERR_GET_ID_FAILURE = HDIO_GET_IDENTITY failed: Invalid argument

I was saying look for both error strings in err, so we are more sure that error we hit is close to error we have taken as identification for non-local drives.

Reply:
Added check for both error strings.